annotation/tests/test_tasks_services.py

```diff
@@ -1,30 +1,240 @@
 from annotation.jobs.services import ValidationSchema
-from annotation.models import File, ManualAnnotationTask
-from annotation.schemas.tasks import ManualAnnotationTaskInSchema
-from annotation.tasks.services import (
-    check_cross_annotating_pages,
-    create_annotation_task,
-    filter_tasks_db,
-    read_annotation_task,
-    read_annotation_tasks,
-    remove_additional_filters,
-    validate_files_info,
-    validate_ids_and_names,
-    validate_task_info,
-    validate_user_actions,
-    validate_users_info,
+from annotation.models import (
+    AnnotatedDoc,
+    AnnotationStatistics,
+    File,
+    ManualAnnotationTask,
 )
+from annotation.schemas.annotations import PageSchema, ParticularRevisionSchema
+from annotation.schemas.tasks import (
+    AgreementScoreComparingResult,
+    AgreementScoreServiceResponse,
+    AnnotationStatisticsInputSchema,
+    ManualAnnotationTaskInSchema,
+    ResponseScore,
+    TaskMetric,
+    TaskStatusEnumSchema,
+)
+
+
+@pytest.fixture
+def mock_task_revisions():
+    mock_revision = MagicMock(spec=AnnotatedDoc)
+    mock_revision.pages = {"1": ["data1"], "2": ["data2"]}
+    mock_revision.failed_validation_pages = [1, 2]
+    mock_revision.validated = [1, 2]
+    yield mock_revision
+
+
+@pytest.fixture
+def mock_metric():
+    metric = MagicMock()
+    metric.task_from = datetime(2024, 1, 1)
+    metric.task_to = datetime(2024, 2, 1)
+    metric.agreement_metric = True
+    yield metric
+
+
+@pytest.fixture
+def mock_stats(mock_task: ManualAnnotationTask, mock_metric: Mock):
+    stat1 = MagicMock()
+    stat1.task = mock_task
+    stat1.task_id = 1
+    stat1.created = datetime(2024, 1, 1, 12, 0, 0)
+    stat1.updated = datetime(2024, 1, 2, 12, 0, 0)
+    stat1.task.user_id = 2
+    stat1.task.file_id = 3
+    stat1.task.pages = [1, 2, 3]
+    stat1.task.status.value = "completed"
+    stat1.task.agreement_metrics = [mock_metric]
+
+    stat2 = MagicMock()
+    stat2.task = mock_task
+    stat2.task_id = 2
+    stat2.created = datetime(2024, 1, 3, 12, 0, 0)
+    stat2.updated = datetime(2024, 1, 4, 12, 0, 0)
+    stat2.task.user_id = 2
+    stat2.task.file_id = 3
+    stat2.task.pages = [1, 2, 3]
+    stat2.task.status.value = "completed"
+    stat2.task.agreement_metrics = [mock_metric]
+
+    stat3 = MagicMock()
+    stat3.task = mock_task
+    stat3.task_id = 3
+    stat3.created = datetime(2024, 1, 5, 12, 0, 0)
+    stat3.updated = datetime(2024, 1, 6, 12, 0, 0)
+    stat3.task.user_id = 2
+    stat3.task.file_id = 3
+    stat3.task.pages = [1, 2, 3]
+    stat3.task.status.value = "completed"
+    stat3.task.agreement_metrics = [mock_metric]
+
+    yield [stat1, stat2, stat3]
+
+
+@pytest.fixture
+def mock_db(mock_stats):
+    mock_db = MagicMock()
+    mock_query = MagicMock()
+    mock_query.filter.return_value.filter.return_value.all.return_value = (
+        mock_stats
+    )
+    mock_db.query.return_value = mock_query
+    return mock_db
+
+
+@pytest.fixture
+def mock_task():
```

Comment:
any reason using mock here? why not the real model?

Reply:
Ok.